bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/DefaultSessionController.java

```diff
@@ -126,6 +141,7 @@ public void modelUpdated(final String modeluri) {
    }
 
    @Override
+   @Deprecated
```

Comment:
Do we really want to deprecate this API? Isn't it still valid V1 API without the Json patch?

Reply:
Yes, we want to deprecate it. This method would only notify V1 clients, even if V2 Clients are listening to the model changes. The new method is the correct implementation, that works with either V1 or V2, or both. V1 implementations should use the new method too (Maybe with an empty supplier for the JsonPatch, if for whatever reason, a V1 Server already migrates to this V2 version of the Model Server API, without actually implementing V2).

The idea is that the Server should support V1 **and** V2 as much as possible. Clients are free to use either V1 or V2 APIs, and they don't need to know if the Server is a V1 or a V2 server (Because a V2 Server stills supports V1, at least for now).

Reply:
Hmm... I see though I might argue a bit differently that if the server should support both versions, adopters that already use that method should not necessarily get a deprecation warning since both calls actually still work but I might be missing something. Hopefully we can deprecate v1 in the near future altogether. Since the PR is already quite big and this is just a minor thing, I am fine with leaving it as is.